spec/OpenLineage.yml

```diff
@@ -0,0 +1,268 @@
+          items:
+            $ref: '#/components/schemas/Dataset'
+        outputs:
+          description: The set of **output** datasets.
+          type: array
+          items:
+            $ref: '#/components/schemas/Dataset'
+        producer:
+          description: URI identifying the producer of this metadata. For example this could be a git url with a given tag or sha
+          type: string
+          format: uri
+          example: https://github.com/OpenLineage/OpenLineage/blob/v1-0-0/client
+      required:
+        - run
+        - job
+        - eventTime
+        - producer
+
+    Run:
+      type: object
+      properties:
+        runId:
+          description: The id of the run, unique relative to the job
+          type: string
+        facets:
+          description: The run facets.
+          type: object
+          properties:
+            nominalTime:
+              $ref: '#/components/schemas/NominalTimeRunFacet'
+            parent:
+              $ref: '#/components/schemas/ParentRunFacet'
+          additionalProperties:
+            $ref: '#/components/schemas/CustomFacet'
+      required:
+        - runId
+
+    Job:
+      type: object
+      properties:
+        namespace:
+          description: The namespace containing that job
+          type: string
+          example: "my-scheduler-namespace"
+        name:
+          description: The unique name for that job within that namespace
+          type: string
+          example: "myjob.mytask"
+        facets:
+          description: The job facets.
+          type: object
+          properties:
+            documentation:
+              $ref: '#/components/schemas/DocumentationJobFacet'
+            sourceCodeLocation:
+              $ref: '#/components/schemas/SourceCodeLocationJobFacet'
+            sql:
+              $ref: '#/components/schemas/SQLJobFacet'
+          additionalProperties:
+            $ref: '#/components/schemas/CustomFacet'
+      required:
+        - namespace
+        - name
+
+    Dataset:
+      type: object
+      properties:
+        namespace:
+          description: The namespace containing that dataset
+          type: string
+          example: "my-datasource-namespace"
+        name:
+          description: The unique name for that dataset within that namespace
+          type: string
+          example: "instance.schema.table"
+        facets:
+          description: The facets for this dataset
+          type: object
+          properties:
+            documentation:
+              $ref: '#/components/schemas/DocumentationDatasetFacet'
+            schema:
+              $ref: '#/components/schemas/SchemaDatasetFacet'
+            dataSource:
+              $ref: '#/components/schemas/DatasourceDatasetFacet'
+          additionalProperties:
+            $ref: '#/components/schemas/CustomFacet'
+      required:
+       - namespace
+       - name
+
+
+# Custom facets
+    CustomFacet:
+      type: object
+      properties:
```

Comment:
Maybe we could consider creating a `data` property, rather than leveraging `additionalProperties` at the root-level of a `CustomFacet`. In my experience, mixing reserved fields and arbitrary user-supplied fields can be really messy - it can make it challenging to add new reserved fields into the mix in the future, as they might conflict with field names that are already utilized in practice.

Reply:
that's a great point.
Another mechanism could be to have a special prefix for those reserved field that can not be used in the user defined schemas. That would alleviate the extra nesting of having a `data` field to nest those. I'll submit an update along those lines.